src/traits.rs

```diff
@@ -1136,30 +1136,42 @@ macro_rules! impl_to_primitive_float_to_float {
     )
 }
 
+macro_rules! impl_to_primitive_float_to_int_or_uint {
+    ($SrcT:ident, $DstT:ident, $slf:expr) => (
+        {
+            if $slf >= $DstT::MIN as $SrcT && $slf <= $DstT::MAX as $SrcT {
```

Comment:
Hmm, how does `as` do rounding/truncation?  I'm concerned that this might reject cases that would have converted to `MIN` or `MAX` just fine, like perhaps `MAX + 0.1`.

(Whether we should do rounding at all is a separate question, but for now we do convert fractional values that are in range of the integer.)